### PR TITLE
Added setup requirements

### PR DESCRIPTION
**Pull request by OrkoHunter**
This adds python setup.py and .gitignore

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+doc/source/reference/generated/*
+doc/source/examples/*
+
+# Generated when 'python setup_egg.py'
+*.egg-info/
+
+# Generated when 'python setup.py build_clib'
+# and 'python setup.py build-ext --inplace'
+networkx_metis/_metis.so
+networkx_metis/_metis.c
```
Comment (ysitu) [inline .gitignore]: The directory is gone.

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+from glob import glob
+import os
+import sys
+
+from setuptools import setup, Extension
+try:
+    from Cython.Build import cythonize
+except ImportError:
+    cythonize = None
```
Comment (ysitu) [inline setup.py]: Do not ignore this error. Cython must be installed.

File: setup.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+if sys.version_info[:2] < (2, 6):
+    print("networkx-metis requires Python version 2.6 or later (%d.%d detected)." %
+          sys.version_info[:2])
+    sys.exit(-1)
+
+packages=["."]
+
+if cythonize:
+    libraries = [('gklib',
+                  {'sources': glob('src/GKlib/*.c'),
```
Comment (ysitu) [inline setup.py]: You also need `glob('src/GKlib/*.cc')`.

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                   'depends': glob('src/GKlib/*.h'),
+                   'include_dirs': ['src/GKlib']}),
+                 ('metis',
+                  {'sources': glob('src/libmetis/*.c'),
```
Comment (ysitu) [inline setup.py]: Add `glob('src/libmetis/*.cc')`.

**Comment by chebee7i** [comment]
Aside: You can use `check-manifest` to help find these things:

```
$ check-manifest -v
listing source files under version control: 93 files and directories
building an sdist: UNKNOWN-0.0.0.tar.gz: 91 files and directories
copying source files to a temporary directory
building a clean sdist: UNKNOWN-0.0.0.tar.gz: 91 files and directories
lists of files in version control and sdist do not match!
missing from sdist:
  src/GKlib/sort.cc
  src/libmetis/gklib.cc
suggested MANIFEST.in rules:
  recursive-include src *.cc
```

After adding the suggestions from @ysitu, it now shows:

listing source files under version control: 93 files and directories
building an sdist: UNKNOWN-0.0.0.tar.gz: 93 files and directories
copying source files to a temporary directory
building a clean sdist: UNKNOWN-0.0.0.tar.gz: 93 files and directories
lists of files in version control and sdist match

Note that the `UNKNOWN` probably needs fixing too.

**Comment by OrkoHunter** [comment]
@chebee7i Yes, thank you for that. `setup.py` and the namespace packaging are yet to be finalized.

File: MANIFEST.in [hunk 5]
```diff
@@ -0,0 +1,13 @@
+# Source files
+include src/GKlib/*.h
+include src/libmetis/*.h
+include src/LICENSE.txt
```
Comment (ysitu) [inline MANIFEST.in]: Do you need to include the `.c` and `.cc` files?

File: .gitignore [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+.coverage
+*.class
+
+# Generated while building documentation.
+doc/source/static/examples/*
+doc/source/reference/generated/*
+doc/source/examples/*
+
+# Generated when 'python setup_egg.py'
+*.egg-info/
```
Comment (ysitu) [inline .gitignore]: Please clean up things that do not apply.

File: setup.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+import sys
+
+from setuptools import setup, Extension
+from Cython.Build import cythonize
+
+if sys.argv[-1] == 'setup.py':
+    print("To install, run 'python setup.py install'")
+    print()
+
+if sys.version_info[:2] < (2, 6):
```
Comment (ysitu) [inline setup.py]: Let's no support Python 2.6. It has EOL'd.

File: setup.py [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+                      'src/libmetis'],
+        libraries=['metis', 'gklib'])])
+else:
+    libraries = None
+    ext_modules = None
+
+if sys.version_info[:2] < (3, 4):
+    install_requires = ['enum34']
+else:
+    install_requires = None
```
Comment (ysitu) [inline setup.py]: `decorator` is also required.

File: MANIFEST.in (hunk 5 above)
Comment (OrkoHunter) [inline MANIFEST.in]: `.c` and `.cc` are being included by `setup.py`. Header files are being missed, so they are included from here.

File: setup.py [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+if sys.argv[-1] == 'setup.py':
+    print("To install, run 'python setup.py install'")
+    print()
+
+if sys.version_info[:2] < (2, 7):
+    print("networkx-metis requires Python version 2.7 or later (%d.%d detected)." %
+          sys.version_info[:2])
+    sys.exit(-1)
+
+if cythonize:
```
Comment (ysitu) [inline setup.py]: Remove this `if`. It has to be true.

File: .gitignore [hunk 10]
```diff
@@ -0,0 +1,17 @@
+*.pyc
+__pycache__
+*~
+
+build/*
+
+# Generated when 'python setup_egg.py'
+*.egg-info/
```
Comment (ysitu) [inline .gitignore]: Remove this also. Wheels is the way to go.

File: MANIFEST.in (hunk 5 above)
Comment (ysitu) [inline MANIFEST.in]: Doesn't `setup.py` already mention the `.h` files?

File: setup.py [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+                      'src/libmetis'],
+        libraries=['metis', 'gklib'])])
+else:
+    libraries = None
+    ext_modules = None
+
+if sys.version_info[:2] < (3, 4):
+    install_requires = ['enum34', 'decorator']
+else:
+    install_requires = None
```
Comment (ysitu) [inline setup.py]: `decorator` is needed in all cases.

File: MANIFEST.in (hunk 5 above)
Comment (OrkoHunter) [inline MANIFEST.in]: Yes, It's mentioned in setup.py under `'depends'` of `libraries`. However the dist package does not include those header files. I used `check-manifest -v` to verify.
```sh
test/ check-manifest -v
listing source files under version control: 103 files and directories
building an sdist: UNKNOWN-0.0.0.tar.gz: 103 files and directories
copying source files to a temporary directory
building a clean sdist: UNKNOWN-0.0.0.tar.gz: 76 files and directories
lists of files in version control and sdist do not match!
missing from sdist:
  src/GKlib/GKlib.h
  src/GKlib/gk_arch.h
  src/GKlib/gk_defs.h
  src/GKlib/gk_externs.h
  src/GKlib/gk_macros.h
  src/GKlib/gk_mkblas.h
  src/GKlib/gk_mkmemory.h
  src/GKlib/gk_mkpqueue.h
  src/GKlib/gk_mkpqueue2.h
  src/GKlib/gk_mkrandom.h
  src/GKlib/gk_mkutils.h
  src/GKlib/gk_proto.h
  src/GKlib/gk_struct.h
  src/GKlib/gk_types.h
  src/GKlib/ms_inttypes.h
  src/GKlib/ms_stat.h
  src/GKlib/ms_stdint.h
  src/libmetis/defs.h
  src/libmetis/gklib_defs.h
  src/libmetis/gklib_rename.h
  src/libmetis/macros.h
  src/libmetis/metis.h
  src/libmetis/metislib.h
  src/libmetis/proto.h
  src/libmetis/rename.h
  src/libmetis/stdheaders.h
  src/libmetis/struct.h
suggested MANIFEST.in rules:
  recursive-include src *.h
```

File: setup.py [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+import os
+import sys
+
+from setuptools import setup, Extension
+from Cython.Build import cythonize
+
+if sys.argv[-1] == 'setup.py':
+    print("To install, run 'python setup.py install'")
+    print()
+
```
Comment (chebee7i) [inline setup.py]: Should you have `from __future__ import print_function` at the top of this file?

Comment (OrkoHunter) [inline setup.py]: Yes, it is needed indeed. However we can also put `print("")` for new line.

File: setup.py [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+ext_modules = cythonize([Extension(
+    '_metis',
+    ['*.pyx'],
+    include_dirs=['src/GKlib',
+                  'src/libmetis'],
+    libraries=['metis', 'gklib'])],
+    extra_compile_args=['-std=c99'])
+
+if sys.version_info[:2] < (3, 4):
+    install_requires = ['enum34', 'decorator']
```
Comment (ysitu) [inline setup.py]: Why did you remove the other branch? It *is* needed. `decorator` should be declared as a dependency whether you use Python 3.4+ or not.

Comment (OrkoHunter) [inline setup.py]: Okay, I changed it.

**Comment by OrkoHunter** [comment]
I added the `name` in `setup.py` as `networkx.addons.metis`. I'll send a follow up PR with the minor tweak in networkx core.
I think we have always discussed the prototype as
```
>>> from networkx.addons import metis
```
But as of this, it's going to be
```
>>> from networkx.addons.metis import metis
```
This is because we can't put all the modules into the `networkx.addons` namespace which in future would raise conflicts with other addons.

**Comment by chebee7i** [comment]
Is this because of the namespace package? E.g. everything is actually put into `networkx/addons/metis/*`?

**Comment by OrkoHunter** [comment]
Yes, *virtually*.

**Comment by chebee7i** [comment]
Honestly, I think that import statement will be confusing to users. But perhaps documentation can help.

**Comment by OrkoHunter** [comment]
Yes that would feel a little confusing for this addon. But, I think it'd more generous to put `networkx-metis` related stuff inside `networkx/addons/metis` rather than just `networkx/addons`.
Yes definitely, documentation will help.

**Comment by chebee7i** [comment]
Just so I'm clear though, if we didn't go with the namespace route, this would not be an issue at all right? Then, users can just do: `from networkx.addons import metis`.

**Comment by OrkoHunter** [comment]
Yes. In that case `metis` would just be a dummy module to import everything from a proper `networkx_metis`.
@chebee7i I think we should step up and use namespace packages. Addons like features take very great use of it. And it's in good use also. I just found [an example](https://github.com/matplotlib/basemap) for matplotlib.
We should resolve the import issue some other way.

**Comment by chebee7i** [comment]
I think I'd be okay doing that, I just don't really understand what is gained by using the namespace package. If you can explain, that would help me quite a bit. But yes, let's just continue b/c that is an easy decision to change if we do at all.

**Comment by chebee7i** [comment]
I guess the value is that people can create addons without having to update `networkx`. Helpful.

**Comment by OrkoHunter** [comment]
Yes , totally agree with you~ And if you consider both the designs too, then the namespace package way looks better than the simple-hackish `metis.py` inside it.

**Comment by OrkoHunter** [comment]
For the namespace package to be installed in `networkx.addons.metis`, don't we need the `__init__.py` with the one liner in `networkx/addons/metis`? I think the one in `networkx/addons` won't suffice.

**Comment by ysitu** [comment]
Wouldn't your `setup.py` install an `__init__.py`?

**Comment by OrkoHunter** [comment]
Even if we don't have it in this package? I've found examples on namespace packages which only deal with top level eg. `zope.interface` or `matplotlib.basemap`. Here we have `networkx.addons.metis` to install stuff in.

**Comment by ysitu** [comment]
Maybe this will help: http://stackoverflow.com/questions/8380381/the-way-to-make-namespace-packages-in-python.

**Comment by chebee7i** [comment]
https://www.python.org/dev/peps/pep-0420/:  "Namespace packages cannot contain an `__init__.py`."

**Comment by ysitu** [comment]
You need this directory structure:

```bash
$ find networkx -name *.py
networkx/__init__.py
networkx/addons/__init__.py
networkx/addons/metis/__init__.py
```

`networkx/__init__.py` and `networkx/addons/__init__.py` each contain just one line:

```python
__import__('pkg_resources').declare_namespace(__name__)
```

I have the following in my dummy `networkx/addons/metis/__init__.py`:

```python
def foo():
  return 3.14
```

Then I can do:

```python
$ python
Python 2.7.6 (default, Mar 22 2014, 22:59:56) 
[GCC 4.8.2] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import networkx.addons.metis
>>> networkx.addons.metis.foo()
3.14
```

**Comment by OrkoHunter** [comment]
@ysitu So, the `networkx/metis/__init__.py` need to be there in the core? Because networkx-metis will probably not install any `__init__.py` there. 

If `networkx/metis/__init__.py` does not contain
```py
__import__('pkg_resources').declare_namespace(__name__)
```
how can `networkx-metis` install it's modules and everything into it?

**Comment by OrkoHunter** [comment]
Oh I see, this directory structure is for `networkx-metis`, isn't it?

**Comment by chebee7i** [comment]
Yes, I think that is correct @OrkoHunter

File: setup.py [hunk 14]
```diff
@@ -0,0 +1,53 @@
...
+               glob('src/libmetis/*.cc'),
+               'depends': glob('src/libmetis/*.h'),
+               'include_dirs': ['src/GKlib',
+                                'src/libmetis']})]
+ext_modules = cythonize([Extension(
+    '_metis',
+    ['*.pyx'],
+    include_dirs=['src/GKlib',
+                  'src/libmetis'],
+    libraries=['metis', 'gklib'])])
```
Comment (ysitu) [inline setup.py]: Reduce the number of newlines in these two assignmentes to make them more readable.

File: setup.py [hunk 15]
```diff
@@ -0,0 +1,53 @@
...
+
+install_requires = ['decorator']
+
+if sys.version_info[:2] < (3, 4):
+    install_requires.append('enum34')
+
+if __name__ == "__main__":
+
+    setup(
+        # name, author, maintainer, author_email, maintainer_email, etc. ??
```
Comment (ysitu) [inline setup.py]: Please fill out the missing fields.

File: .gitignore [hunk 16]
```diff
@@ -0,0 +1,14 @@
+*.pyc
+__pycache__
+*~
+
+build/*
+
+# Generated when 'python setup.py build_clib'
+# and 'python setup.py build-ext --inplace'
+_metis.so
```
Comment (ysitu) [inline .gitignore]: IIRC Python 3 generates a different name than `_metis.so`.

Comment (OrkoHunter) [inline .gitignore]: I'm having a little problem while testing with `python3 setup.py build_ext`, it says `no module named Cython installed`. 
`pip3 install Cython` didn't fix that either. So, `_metis.so` would be a mystery for me, as of now.

Comment (chebee7i) [inline .gitignore]: Are you using a virtualenv? You'd have to verify that Cython is actually installed with the correct Python if not.

Comment (OrkoHunter) [inline .gitignore]: No, I'm not using any virtual env. Actually the same happened with networkx. I had to install it using `python3` to use it from python 3. I thought just one time package installation would be enough to use from both versions of python.

Comment (chebee7i) [inline .gitignore]: pip installing will place the package in a corresponding `pythonX.Y/site-packages/`. So not only is the pip command different, but so is the installation directory (and also the python command).

**Comment by SanketDG** [comment]
Something like using [pyenv](https://github.com/yyuu/pyenv) and [pyenv-virtualenv](https://github.com/yyuu/pyenv-virtualenv) could be helpful, if you want both `python2` and python3` support in your projects.

File: .gitignore (hunk 16 above)
Comment (OrkoHunter) [inline .gitignore]: > IIRC Python 3 generates a different name than _metis.so.

@ysitu Yes, python3 generates `_metis.cpython-34m.so`
`python3-dev` was needed to install cython for python3.

File: .gitignore [hunk 17]
```diff
@@ -0,0 +1,15 @@
...
+_metis.cpython-34m.so
```
Comment (ysitu) [inline .gitignore]: Why not `_metis.*.so`?

File: MANIFEST.in (hunk 5 above)
Comment (OrkoHunter) [inline MANIFEST.in]: I don't feel the need of including `src/*`, `_metis.c`, `_metis.pyx` and `_metis.pxd` in the distribution created by setup.py. Things which are required to be installed are all the python files plus the `.so` file generated through Cython. So maybe, we can omit first 8 lines from the `MANIFEST.in`.

File: setup.py [hunk 18]
```diff
@@ -0,0 +1,64 @@
...
+	 'include_dirs': ['src/GKlib']}),
+	('metis', {'sources': glob('src/libmetis/*.c') + glob('src/libmetis/*.cc'),
+	 'depends': glob('src/libmetis/*.h'),
+	 'include_dirs': ['src/GKlib', 'src/libmetis']})]
+
+ext_modules = cythonize([Extension('networkx.addons.metis._metis', ['*.pyx'],
+                                   include_dirs=['src/GKlib', 'src/libmetis'],
+                                   libraries=['metis', 'gklib'])])
+
+install_requires = ['decorator']
```
Comment (ysitu) [inline setup.py]: `six`?

File: MANIFEST.in (hunk 5 above)
Comment (ysitu) [inline MANIFEST.in]: There is indeed no need to install the C/C++/Cython source files. If that does not cause problems, it is okay to remove them from `MANIFEST.in`.

Comment (OrkoHunter) [inline MANIFEST.in]: Cool. I'm working on Travis CI right now, so mainly installation is what we have now.

**Comment by OrkoHunter** [comment]
Pretty much complete, I guess. Changes might be made because of the issue mentioned in #16.

File: setup.cfg [hunk 19]
```diff
@@ -0,0 +1,2 @@
+[build_ext]
+inplace=1
```
Comment (ysitu) [inline setup.cfg]: Do you need this in the released package? Does it affect installation?

File: setup.py [hunk 20]
```diff
@@ -0,0 +1,64 @@
...
+
+from glob import glob
+import os
+import sys
+
+from setuptools import setup, Extension
+from Cython.Build import cythonize
+
+if sys.argv[-1] == 'setup.py':
+    print("To install, run 'python setup.py install'")
```
Comment (ysitu) [inline setup.py]: Use single quotes for consistency.

File: setup.py [hunk 21]
```diff
@@ -0,0 +1,64 @@
...
+    print()
+
+if sys.version_info[:2] < (2, 7):
+    print("networkx-metis requires Python version 2.7 or later (%d.%d detected)." %
```
Comment (ysitu) [inline setup.py]: Use single quotes for consistency.

File: setup.py [hunk 22]
```diff
@@ -0,0 +1,64 @@
...
+				'Intended Audience :: Developers',
+				'Intended Audience :: Science/Research',
+				'License :: OSI Approved :: Apache License Version 2',
+				'Operating System :: OS Independent',
+				'Programming Language :: Python :: 2',
+				'Programming Language :: Python :: 2.7',
+				'Programming Language :: Python :: 3',
+				'Programming Language :: Python :: 3.2',
+				'Programming Language :: Python :: 3.3',
+				'Programming Language :: Python :: 3.4',],
```
Comment (ysitu) [inline setup.py]: Indentation.

File: setup.py [hunk 23]
```diff
@@ -0,0 +1,64 @@
...
+        maintainer         = 'NetworkX Developers',
+        maintainer_email   = 'networkx-discuss@googlegroups.com',
+        description        = 'NetworkX Addon to allow graph partitioning with METIS',
+        keywords           = ['Networks', 'Graph Theory', 'Mathematics',
+                              'network', 'graph', 'math', 'graph partitioning'],
+        classifiers        = ['Development Status :: 5 - Production/Stable',
+				'Intended Audience :: Developers',
+				'Intended Audience :: Science/Research',
+				'License :: OSI Approved :: Apache License Version 2',
+				'Operating System :: OS Independent',
```
Comment (ysitu) [inline setup.py]: I wish that we could do something testing on Windows.

File: MANIFEST.in [hunk 24]
```diff
@@ -0,0 +1,10 @@
+# Wrapper files
+graft networkx
+
+# Build files
+exclude _metis.c
+
+# Documentaion and License files
+include license.txt
```
Comment (ysitu) [inline MANIFEST.in]: `LICENSE.txt`

File: setup.py [hunk 25]
```diff
@@ -0,0 +1,64 @@
...
+          sys.version_info[:2])
+    sys.exit(-1)
+
+libraries = [
+	('gklib', {'sources': glob('src/GKlib/*.c') + glob('src/GKlib/*.cc'),
+	 'depends': glob('src/GKlib/*.h'),
+	 'include_dirs': ['src/GKlib']}),
+	('metis', {'sources': glob('src/libmetis/*.c') + glob('src/libmetis/*.cc'),
+	 'depends': glob('src/libmetis/*.h'),
+	 'include_dirs': ['src/GKlib', 'src/libmetis']})]
```
Comment (ysitu) [inline setup.py]: Indentation.

File: setup.cfg (hunk 19 above)
Comment (OrkoHunter) [inline setup.cfg]: Yes pretty much. Without this, `python setup.py install` will leave the `.so` file in the `build` directory, so installing will become a two step 'build --inplace` and 'install` process.

With this, `python setup.py install` will alone do the job.

File: setup.py (hunk 23 above)
Comment (OrkoHunter) [inline setup.py]: I actually have both Linux and Windows. So, I'll look into it after we are done with Linux.

File: setup.py [hunk 26]
```diff
@@ -0,0 +1,64 @@
...
+                 'Intended Audience :: Developers',
+                 'Intended Audience :: Science/Research',
+                 'License :: OSI Approved :: Apache License Version 2',
+                 'Operating System :: OS Independent',
+                 'Programming Language :: Python :: 2',
+                 'Programming Language :: Python :: 2.7',
+                 'Programming Language :: Python :: 3',
+                 'Programming Language :: Python :: 3.2',
+                 'Programming Language :: Python :: 3.3',
+                 'Programming Language :: Python :: 3.4',],
```
Comment (ysitu) [inline setup.py]: ? You need to align these lines with `'Development Status...`'.

Comment (SanketDG) [inline setup.py]: Like this:
```python
['Development Status :: 5 - Production/Stable',
 'Intended Audience :: Developers',
 'Intended Audience :: Science/Research',
 'License :: OSI Approved :: Apache License Version 2',
 'Operating System :: OS Independent',
 'Programming Language :: Python :: 2',
 'Programming Language :: Python :: 2.7',
 'Programming Language :: Python :: 3',
 'Programming Language :: Python :: 3.2',
 'Programming Language :: Python :: 3.3',
 'Programming Language :: Python :: 3.4',],

File: setup.py [hunk 27]
```diff
@@ -0,0 +1,64 @@
...
+from setuptools import setup, Extension
+from Cython.Build import cythonize
+
+if sys.argv[-1] == 'setup.py':
+    print('To install, run \'python setup.py install\'')
+    print()
+
+if sys.version_info[:2] < (2, 7):
+    print('networkx-metis requires Python version 2.7 or later (%d.%d detected).' %
+          sys.version_info[:2])
```
Comment (SanketDG) [inline setup.py]: use `str.format`?

File: setup.py (hunk 26 above)
Comment (OrkoHunter) [inline setup.py]: Yes, got it. `pep8` wasn't showing any errors on the previous indentation so, I had not changed it.

File: setup.py (hunk 27 above)
Comment (OrkoHunter) [inline setup.py]: They seem to work in case of tuples.

File: setup.py [hunk 28]
```diff
@@ -0,0 +1,64 @@
...
+                              'License :: OSI Approved :: Apache License Version 2',
+                              'Operating System :: OS Independent',
+                              'Programming Language :: Python :: 2',
+                              'Programming Language :: Python :: 2.7',
+                              'Programming Language :: Python :: 3',
+                              'Programming Language :: Python :: 3.2',
+                              'Programming Language :: Python :: 3.3',
+                              'Programming Language :: Python :: 3.4',],
+        packages           = ['networkx.addons.metis', 'networkx.addons', 'networkx'],
+        namespace_packages = ['networkx.addons.metis', 'networkx.addons', 'networkx'],
```
Comment (ysitu) [inline setup.py]: All looks good. One final question: does `networkx.addons.metis` need to be a namespace package?

File: setup.py (hunk 27 above)
Comment (SanketDG) [inline setup.py]: Do you mean it doesn't work? Note that you have to unpack the tuple, since `format` is a method:
```python
print('networkx-metis requires Python version 2.7 or later ({}.{} detected).'.format(*sys.version_info[:2]))
```

Comment (OrkoHunter) [inline setup.py]: Sorry for being vague @SanketDG. I meant `%` seems to work in case of tuples.

File: setup.py (hunk 28 above)
Comment (OrkoHunter) [inline setup.py]: I tried removing it and keeping 
```
namespace_packages = ['networkx.addons', 'networkx']
```
This gives the freedom of putting a `__init__.py` inside `networkx/addons/metis/` which is also *necessary* because it will never be recognized as a package (because it won't have a counter-part in networkx core like networkx/addons).

So, with this configuration in `networkx-metis` and https://github.com/networkx/networkx/pull/1595 in `networkx` I tried making an import from `networkx.addons.metis` and it worked fine. (I imported `enums` because it's independent of `_metis.so`)

So, I should update the changes I mentioned.

File: setup.py (hunk 27 above)
Comment (SanketDG) [inline setup.py]: Yes, but `str.format` looks pretty and is much more readable. And `%` works because it unpacks the tuple by default, since you have two `%`s. If you try to print the tuple as a whole with just one %, it will fail. This is more reason on why to use `str.format` to avoid this strange behaviour.

File: setup.py (hunk 28 above)
Comment (ysitu) [inline setup.py]: I am confused. Do you mean that `networkx.addons.metis` needs to be turned into a normal package?

File: setup.py (hunk 27 above)
Comment (OrkoHunter) [inline setup.py]: Yes, you are right. I just read somewhere the last part you mentioned and it's convincing. I'll make the modification. Thank you.

File: setup.py (hunk 28 above)
Comment (OrkoHunter) [inline setup.py]: In this case, yes. If we don't put the `__init__.py`, we will need the `one-liner __init__.py` in the networkx core, inside `networkx/addons/metis` directory, which will cause an unwanted directory inside `networkx/addons/` in the networkx core.

Comment (OrkoHunter) [inline setup.py]: Or maybe I am confused. I'm trying to make things clear, give me a moment please.

File: setup.py [hunk 29]
```diff
@@ -0,0 +1,65 @@
...
+                              'License :: OSI Approved :: Apache License Version 2',
+                              'Operating System :: OS Independent',
+                              'Programming Language :: Python :: 2',
+                              'Programming Language :: Python :: 2.7',
+                              'Programming Language :: Python :: 3',
+                              'Programming Language :: Python :: 3.2',
+                              'Programming Language :: Python :: 3.3',
+                              'Programming Language :: Python :: 3.4',],
+        packages           = ['networkx.addons.metis', 'networkx.addons', 'networkx'],
+        namespace_packages = ['networkx.addons', 'networkx'],
```
Comment (OrkoHunter) [inline setup.py]: `networkx.addons.metis` was not a namespace package. Removed.
